lib/kernel.coffee

```diff
@@ -496,3 +494,5 @@ class Kernel
 
         unless @onlyConnect
             @kernelProcess.kill 'SIGKILL'
+
+        @onDestroy(this)
```

Comment:
The current kernel doesn't need `onDestroy` because it is killed by calling KernelManager (now ZMQGateway), not the kernel itself.

I am changing this such that calling the kernel's destroy directly is sufficient. To do this the kernel must know how to unlist itself from the ZMQGateway, which requires either a pointer to the ZMQGateway or a callback.